Dev cleanup

yelidrissi opened the pull request.

robert-ngo committed:
Clean up Make file, use docker-compose, add apache server

yelidrissi committed:
Attempt to make Robert's php server for the GDocs-Export add-on functionally equivalent to the older Node.js server.

Many changes were made across the project in order to achieve this.

dergachev commented on .env:
```diff
@@ -0,0 +1,4 @@
+PORT_PREFIX=80
+DOCKER_IMAGE=robert_gdocs
```
Fix prefix, perhaps gdocs_server

dergachev commented on Dockerfile:
```diff
@@ -1,19 +1,7 @@
-FROM  ubuntu:trusty
-MAINTAINER Alex Dergachev <alex@evolvingweb.ca>
+FROM  dergachev/gdocs-export:latest
```
you are killing my Dockerfile that builds dergachev/gdocs-export image. Problematic.

dergachev commented on Dockerfile:
```diff
@@ -22,46 +10,38 @@ RUN apt-get install -y locales
-COPY server/envvars /etc/apache2/envvars
+COPY server/envvars /etc/apache2/envvars.bak
+RUN cat /etc/apache2/envvars.bak | tr -s '\r' '\n' > /etc/apache2/envvars
+RUN ["/bin/bash","-c","source /etc/apache2/envvars"]
```
@robert please verify necessity

dergachev commented on Dockerfile:
```diff
@@ -22,46 +10,38 @@ RUN apt-get install -y locales
+
+
+
+RUN service apache2 restart
```
quite sure not necessary

dergachev commented on Dockerfile:
```diff
@@ -22,46 +10,38 @@ RUN apt-get install -y locales
+EXPOSE 12736
+WORKDIR /var/gdocs-export/
+
+ARG extra
```
@Robert please confirm this makes sense

dergachev commented on Makefile:
```diff
@@ -5,15 +5,15 @@ include .env
 
 #DATE=$(eval DATE_DIR=$(shell date +%Y-%m))
-DATE=$(shell date +%Y-%m)
+#DATE=$(shell date +%Y-%m)
```
In the future, consider storing build files in folders named after user id.

dergachev commented on lib/pandoc-filter.py.bak:
```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
```
Remove this .bak file, it's redundant

dergachev commented on lib/pandoc-filter.py:
```diff
@@ -1,43 +1,31 @@
 #!/usr/bin/env python
```
separate PR please for this change

dergachev commented on server/envvars.bak:
```diff
@@ -0,0 +1,47 @@
+# envvars - default environment variables for apache2ctl
```
remove this file

dergachev commented on server/www/index.php:
```diff
@@ -9,6 +9,81 @@
 include('includes/Message.php');
```
get rid of Message.php if it's not used

dergachev commented on server/www/index.php:
```diff
@@ -9,6 +9,81 @@
+}
+
+function upload() {
+    if(!isset($_REQUEST['id']) || !isset($_REQUEST['token'])) endit(400, "Bad Request.");
```
run through drupal code sniffer (drupalcs)

dergachev reviewed (commented):
Here's my review

yelidrissi committed:
Implementing the tasks assigned during the pull-request review.

The tasks/comments/reviews as they exist here:
https://github.com/dergachev/gdocs-export/pull/12

Tasks for Robert weren't dealt with.

yelidrissi committed:
Delete envvars.bak

yelidrissi committed:
Delete pandoc-filter.py.bak

yelidrissi commented:
The tasks assigned during the review process have been dealt with.

yelidrissi committed:
Delete token.json

yelidrissi committed:
Delete credentials.json

robert-ngo commented on Dockerfile:
```diff
@@ -32,16 +70,13 @@ RUN ["/bin/bash","-c","source /etc/apache2/envvars"]
 CMD apachectl -D FOREGROUND
 
 # allow gdocs to run script as root
 RUN printf "\ngdocs ALL=(root) NOPASSWD: /var/gdocs-export/server/scripts/web-convert-gdoc.sh\n" >> /etc/sudoers
```
Should remove this line
